src/test/java/org/wso2/carbon/connector/integration/test/StackExchangeConnectorIntegrationTest.java

```diff
@@ -0,0 +1,699 @@
+                        .queryParam("pagesize", "3").build();
+        return getStackExchangeItems(tagUrl);
+    }
+
+    /* ============================================= Filter routes ============================================= */
+
+    private StackExchangeItems getStackExchangeFilterItems(String filterName) throws Exception {
+
+        StackExchangeUrl filterUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/filters/" + filterName).build();
+        return getStackExchangeItems(filterUrl);
+    }
+
+    /* ============================================= Question routes ============================================= */
+
+    private StackExchangeItems getStackExchangeQuestionItems() throws Exception {
+
+        StackExchangeUrl questionUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/search/advanced")
+                        .queryParam("site", site)
+                        .queryParam("accepted", "False")
+                        .queryParam("answers", "1").build();
+        return getStackExchangeItems(questionUrl);
+    }
+
+    /* ============================================= Answer routes ============================================= */
+
+    private StackExchangeItems getStackExchangeAnswerItems(int questionId, String site) throws Exception {
+
+        StackExchangeUrl answerUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/questions/" + questionId + "/answers")
+                        .queryParam("site", site).build();
+        return getStackExchangeItems(answerUrl);
+    }
+
+    private StackExchangeItems getStackExchangeAnswerItems(int questionId) throws Exception {
+
+        StackExchangeUrl answerUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/questions/" + questionId + "/answers")
+                        .queryParam("site", site).build();
+        return getStackExchangeItems(answerUrl);
+    }
+
+    /* ============================================= Privilege routes ============================================= */
+
+    private StackExchangeItems getStackExchangePrivilegeItems(String accessToken) throws Exception {
+
+        StackExchangeUrl privilegeUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/me/privileges/")
+                        .queryParam("site", site)
+                        .queryParam("key", key)
+                        .queryParam("access_token", accessToken).build();
+        return getStackExchangeItems(privilegeUrl);
+    }
+
+    private <T> void setListLikePropertyInSystem(String propertyName, T[] array) {
+
+        stackExchangeProperties.setProperty(propertyName, getTArrayAsSemicolonDelimitedString(array));
+    }
+
+    private <T> void setListLikePropertyInConnector(String propertyName, T[] array) {
+
+        connectorProperties.setProperty(propertyName, getTArrayAsSemicolonDelimitedString(array));
+    }
+
+    private <T> String getTArrayAsSemicolonDelimitedString(T[] array) {
```

Comment:
Add method level comments and check this throughout the PR

Reply:
Fix with 004a93e